R/derive_extreme_records.R

```diff
@@ -1,10 +1,54 @@
 #' Add the First or Last Observation for Each By Group as New Records
 #'
-#' Add the first or last observation for each by group as new observations. It
-#' can be used for example for adding the maximum or minimum value as a separate
-#' visit. All variables of the selected observation are kept. This distinguish
-#' `derive_extreme_records()` from `derive_summary_records()`, where only the by
-#' variables are populated for the new records.
+#' Add the first or last observation for each by group as new observations. The
+#' new observations can be selected from the input dataset or an additional
+#' dataset. It can be used for example for adding the maximum or minimum value
+#' as a separate visit. All variables of the selected observation are kept. This
+#' distinguishes `derive_extreme_records()` from `derive_summary_records()`,
+#' where only the by variables are populated for the new records.
+#'
+#' @param dataset Input dataset
+#'
+#'   If `dataset_add` is not specified, the new records are selected from the
+#'   input dataset. In this case the variables specified by `by_vars` and
+#'   `order` are expected.
+#'
+#' @param dataset_ref Reference dataset
+#'
+#'   The variables specified for `by_vars` are expected. For each
+#'   observation of the specified dataset a new observation is added to the
+#'   input dataset.
+#'
+#' @param dataset_add Additional dataset
+#'
+#'   Observations from the specified dataset are added as new records to the
+#'   input dataset (`dataset`).
+#'
+#'   All observations in the specified dataset fulfilling the condition
+#'   specified by `filter_source` are considered. If `mode` and `order` are
+#'   specified, the first or last one within each by group defined by `by_vars`
+#'   is selected.
+#'
+#'   If the argument is not specified, the input dataset (`dataset`) is used.
+#'
+#'   The variables specified by the `by_vars` and `order` argument (if
+#'   applicable) are expected.
+#'
+#' @param by_vars Grouping variables
+#'
+#'   If `dataset_ref` is specified, this argument must be specified.
+#'
+#'   *Permitted Values*: list of variables created by `exprs()`
+#'
+#' @param filter_add Filter for additional dataset (`dataset_add`)
+#'
+#'   Only observations in `dataset_add` fulfilling the specified condition are
+#'   considered.
+#'
+#'   For by groups with at least one observation `exist_flag` is set to
+#'   `true_value`.
+#'
+#'   For all other by groups `exist_flag` is set to `false_value`.
```

Comment:
I am finding this a little confusing with the reference to the `exist_flag` argument being referenced.  Also, are `true_value` and `false_values` just `Y/N`?   If so, should we specifcy that is the output or a can a user specify the values? 
![image](https://user-images.githubusercontent.com/10111024/236565312-13c7835a-b04d-4e32-a442-87b74ae99906.png)

Reply:
ah I see it can be specified.  I think we should highlight that exisit_flag, true_value, false_value are arguments in the function in the `filter_add` argument.  I got a little confused.  :)

Reply:
Updated, is it clearer now?